app/AvailablePlugin/PrivacyIdeaAuthenticator/Controller/TotpTokensController.php

```diff
@@ -161,8 +161,14 @@ function checkDeleteDependencies($curdata) {
       throw new InvalidArgumentException(_txt('er.notprov.id', array(_txt('pl.privacyideaauthenticator.fd.serial'))));
     }
 
-    $this->PrivacyIdea->deleteToken($this->viewVars['vv_authenticator']['PrivacyIdeaAuthenticator'],
+    $return_response = $this->PrivacyIdea->deleteToken($this->viewVars['vv_authenticator']['PrivacyIdeaAuthenticator'],
                                     $curdata['TotpToken']['serial']);
+    if(!$return_response->result->status) {
```

Comment:
I'm not sure if this walking the array is always guaranteed to be set, but if not a better test is something like

`if(!isset($return_response->result->status) || !$return_response->result->status)`

Dereferencing a non-existant variable will throw a stack trace, but checking `isset()` first will prevent that.